### PR TITLE
[FIX JENKINS-44176] Fix full display name when an organization is selected

### PR DESCRIPTION
**Pull request by alvarolobato**
# Description

See [JENKINS-44176](https://issues.jenkins-ci.org/browse/JENKINS-44176).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by alvarolobato** [comment]
@reviewbybees

File: blueocean-dashboard/src/main/js/components/PipelineRowItem.jsx [hunk 1]
```diff
@@ -7,14 +7,16 @@ import { capable, UrlConfig } from '@jenkins-cd/blueocean-core-js';
 import { MATRIX_PIPELINE } from '../Capabilities';
 import { Icon } from '@jenkins-cd/react-material-icons';
 
-function generateRedirectLink(pipeline) {
+function generateRedirectLink(pipeline, organization) {
     if (capable(pipeline, MATRIX_PIPELINE)) {
+        const fullDisplayPath = organization ? `${organization}/${pipeline.fullDisplayName}` : pipeline.fullDisplayName;
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/components/PipelineRowItem.jsx]: ... only thing, is this the display name of the organization; seems inconsistent with the `pipeline` object, maybe rename it to `organizationDisplayName` to be clear....

Comment (alvarolobato) [inline blueocean-dashboard/src/main/js/components/PipelineRowItem.jsx]: It's for consistency with line 73 (https://github.com/jenkinsci/blueocean-plugin/pull/1054/files#diff-739e1611c7dd82ec706a63ac6c6fcbfcR73) isn't that the same?

Comment (kzantow) [inline blueocean-dashboard/src/main/js/components/PipelineRowItem.jsx]: Good point, it's poorly named there, too :)

Comment (kzantow) [inline blueocean-dashboard/src/main/js/components/PipelineRowItem.jsx]: Only a minor NIT

File: blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/PipelineSearch.java [hunk 2]
```diff
@@ -93,7 +93,7 @@ public String getType() {
             items = getAllItems(org);
         }
         items = ContainerFilter.filter(items);
-        final Iterator<BluePipeline> pipelineIterator = new PipelineContainerImpl()
+        final Iterator<BluePipeline> pipelineIterator = new PipelineContainerImpl(org)
```
Comment (vivek) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/PipelineSearch.java]: 👍

File: blueocean-rest-impl/src/test/java/io/jenkins/blueocean/service/embedded/PipelineApiTest.java [hunk 3]
```diff
@@ -705,4 +710,70 @@ public void actionsTest() throws Exception {
         Assert.assertEquals(2, actions.size());
     }
 
+    /**
+     * Tests the API when the organization root is a folder instead of the root jenkins. That organization should only
+     * see the elements under that folder.
+     */
+    @Test
+    public void testOrganizationFolder() throws IOException, ExecutionException, InterruptedException {
```
Comment (vivek) [inline blueocean-rest-impl/src/test/java/io/jenkins/blueocean/service/embedded/PipelineApiTest.java]: Thanks for adding test 👍

**Comment by alvarolobato** [comment]
@reviewbybees done

**Comment by kzantow** [comment]
Running ATH here: https://ci.blueocean.io/blue/organizations/jenkins/ATH-Jenkinsfile/detail/master/2397/pipeline

**Comment by i386** [comment]
ATH passes @alvarolobato - ill merge this in.

**Comment by alvarolobato** [comment]
Thanks for the quick review.

**Comment by i386** [comment]
@alvarolobato you are welcome 🎉

**Pull request merged**